dm/master/server.go

```diff
@@ -170,11 +170,8 @@ func (s *Server) Start(ctx context.Context) (err error) {
 	if err != nil {
 		return terror.ErrMasterTLSConfigNotValid.Delegate(err)
 	}
-	if tls2 != nil && tls2.TLSConfig() != nil {
-		tls2.TLSConfig().InsecureSkipVerify = true
```

Comment:
if `InsecureSkipVerify` is true, the  argument `verifiedChains` in `VerifyPeerCertificate` will be nil, so will return error here https://github.com/pingcap/tidb-tools/blob/03575b185007b81384dc0728020cf5a95cd28f9e/pkg/utils/security.go#L99

more details refer to https://golang.org/src/crypto/tls/common.go?s=17676:27385#L556

Reply:
If don't set `InsecureSkipVerify`, "client verifies the server's certificate chain and host name". Did we need extra certs to pass this verify? @lichunzhu

Reply:
No, I don't think so. Why will we need extra certs?

Reply:
> No, I don't think so. Why will we need extra certs?

Because we set `InsecureSkipVerify` before this PR, I think that behaviour should come from some reason, such as we can't set CA or something

Reply:
Actually from the comment this argument should be used only for test. PD doesn't use this argument but can still works. Maybe after this we need to set the correct URL when we publish CA certs.
https://docs.pingcap.com/zh/tidb-in-kubernetes/stable/enable-tls-between-components#%E4%BD%BF%E7%94%A8-cfssl-%E7%B3%BB%E7%BB%9F%E9%A2%81%E5%8F%91%E8%AF%81%E4%B9%A6

Reply:
I also use InsecureSkipVerify [here](https://github.com/pingcap/dm/blob/255a9c2be64a891d45ad62e0c9f228e191f155f2/syncer/syncer.go#L2652) in #899, any idea?

Reply:
> I also use InsecureSkipVerify [here](https://github.com/pingcap/dm/blob/255a9c2be64a891d45ad62e0c9f228e191f155f2/syncer/syncer.go#L2652) in #899, any idea? If not set, binlog streamerController will return same error `transport: authentication handshake failed: tls: either ServerName or InsecureSkipVerify must be specified in the tls.Config`

I found this problem maybe caused by https://github.com/pingcap/tidb-tools/blob/master/pkg/utils/security.go#L73 , `tls.Config` has a `ServerName` field which says 

```
	// ServerName is used to verify the hostname on the returned
	// certificates unless InsecureSkipVerify is given. It is also included
	// in the client's handshake to support virtual hosting unless it is
	// an IP address.
```